sql/hive/src/test/scala/org/apache/spark/sql/hive/StatisticsSuite.scala

```diff
@@ -218,6 +224,49 @@ class StatisticsSuite extends StatisticsCollectionTestBase with TestHiveSingleto
     }
   }
 
+  test("update/eliminate column-level stats") {
+    val table = "update_col_stats_table"
+    withTable(table) {
+      sql(s"CREATE TABLE $table (c1 INT, c2 STRING, c3 DOUBLE)")
+      sql(s"ANALYZE TABLE $table COMPUTE STATISTICS FOR COLUMNS c1")
+      // Don't keep column stats for tables without any records.
```

Comment:
but we do keep, and you tested it...  https://github.com/apache/spark/pull/18292/files#diff-1c464c86b68c2d0b07e73b7354e74ce7R235